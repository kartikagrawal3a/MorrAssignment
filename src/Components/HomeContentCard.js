import React from 'react';
import { RiClapperboardFill } from "react-icons/ri";
import HomeContentCSS from '../styles/HomeContent.module.css';

const HomeContentCard = (props) => {
    return(
        <>
            <div title="main">
                <div className={HomeContentCSS.card} title="card">
                    <RiClapperboardFill className={HomeContentCSS.bg} title="media" />
                    <p title="type">{props.type}</p>
                </div>
                <p className="info" title="typeName">{`Popular ${props.type}`}</p>
            </div>
        </>
    );
}

export default HomeContentCard;