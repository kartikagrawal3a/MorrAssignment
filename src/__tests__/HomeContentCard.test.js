import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import HomeContentCard from '../Components/HomeContentCard';

afterEach(()=>{
    cleanup();
})

test("checkHeaderRender1", ()=>{
    const cat = "Movies";
    const { getByTitle} = render(<HomeContentCard type = {cat} />);
    getByTitle("main");
    getByTitle("card");
    getByTitle("media");
    const type = getByTitle("type");
    expect(type).toHaveTextContent("Movies")
    const typeName = getByTitle("typeName");
    expect(typeName).toHaveTextContent("Popular Movies")
})

test("checkHeaderRender2", ()=>{
    const cat = "Series";
    const { getByTitle} = render(<HomeContentCard type = {cat} />);
    getByTitle("main");
    getByTitle("card");
    getByTitle("media");
    const type = getByTitle("type");
    expect(type).toHaveTextContent("Series")
    const typeName = getByTitle("typeName");
    expect(typeName).toHaveTextContent("Popular Series")
})

test('match snapshot #1', ()=>{
    const cat = "Movies";
    const tree = renderer.create(<HomeContentCard type = {cat} />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('match snapshot #2', ()=>{
    const cat = "Series";
    const tree = renderer.create(<HomeContentCard type = {cat} />).toJSON();
    expect(tree).toMatchSnapshot();
})