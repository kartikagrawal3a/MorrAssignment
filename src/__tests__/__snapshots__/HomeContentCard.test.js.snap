// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot #1 1`] = `
<div
  title="main"
>
  <div
    className="card"
    title="card"
  >
    <svg
      className="bg"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        media
      </title>
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M17.998 7l2.31-4h.7c.548 0 .992.445.992.993v16.014a1 1 0 0 1-.992.993H2.992A.993.993 0 0 1 2 20.007V3.993A1 1 0 0 1 2.992 3h3.006l-2.31 4h2.31l2.31-4h3.69l-2.31 4h2.31l2.31-4h3.69l-2.31 4h2.31z"
        />
      </g>
    </svg>
    <p
      title="type"
    >
      Movies
    </p>
  </div>
  <p
    className="info"
    title="typeName"
  >
    Popular Movies
  </p>
</div>
`;

exports[`match snapshot #2 1`] = `
<div
  title="main"
>
  <div
    className="card"
    title="card"
  >
    <svg
      className="bg"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        media
      </title>
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M17.998 7l2.31-4h.7c.548 0 .992.445.992.993v16.014a1 1 0 0 1-.992.993H2.992A.993.993 0 0 1 2 20.007V3.993A1 1 0 0 1 2.992 3h3.006l-2.31 4h2.31l2.31-4h3.69l-2.31 4h2.31l2.31-4h3.69l-2.31 4h2.31z"
        />
      </g>
    </svg>
    <p
      title="type"
    >
      Series
    </p>
  </div>
  <p
    className="info"
    title="typeName"
  >
    Popular Series
  </p>
</div>
`;
